lib/parsec/validate.py

```diff
@@ -379,18 +380,9 @@ def strip_and_unquote_list(cls, keys, value):
         Return (list):
             Processed value as a list.
         """
-        if value.startswith('"'):
-            # double-quoted values
-            match = cls._REC_DQV.match(value)
-            if match:
-                value = match.groups()[0]
-            values = cls._REC_DQ_L_VALUE.findall(value)
-        elif value.startswith("'"):
-            # single-quoted values
-            match = cls._REC_SQV.match(value)
-            if match:
-                value = match.groups()[0]
-            values = cls._REC_SQ_L_VALUE.findall(value)
+        if value.startswith('"') or value.startswith("'"):
+            lexer = shlex.shlex(value, posix=True)
+            values = [t for t in lexer if t != ","]
```

Comment:
Does it mean that lists can now be space-delimited?

I used `shlex` for parsing most list-like values in Rose configuration (and Perl's Text::ParseWords::shellwords in FCM configuration before that), so I strongly support moving towards this approach.

Reply:
Ooohh, that's a unplanned ... feature/bug? I updated the test locally-only, to see if it worked without the comma, with just one space. And it worked.

Not sure if that's all right?

Reply:
Good spotting. That'll break some suites as quoting is optional.
```
handlers = echo hello %(event)s, echo goodbye %(event)s
```

Reply:
Let me see if that's fixable with `shlex`, or if we should just drop this option. Really good spotting! Will try to add a test to make sure this behaviour is covered.

Reply:
(This is when I think we should move to YAML or TOML or configuration format with native support for lists/arrays.)

Reply:
> ~~YAML or TOML~~ Python

Sure but do you really think *most users* will start writing suites-as-Python-programs, when we have a Python API for that?  If not, our config file format is still very much in view of everyone. Nice if it was a "standard" format.

Reply:
Sure but do you really think that *most users* would change to a new Cylc configuration file format?

Reply:
Touche!  Unlike for the Python API, they would have to switch: I suggested (in the email) that we only supply a YAML converter at cylc-8.0, after which we do not support `suite.rc` at all.  We certainly don't want to maintain two distinct config file formats in perpetuity.